PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py

```diff
@@ -172,6 +172,56 @@ def miniAOD_customizeCommon(process):
     del process.slimmedMETsNoHF.caloMET
     # ================== NoHF pfMET
 
+    #  ==================  CHSMET 
+    process.CHSCands = cms.EDFilter("CandPtrSelector",
+                                    src=cms.InputTag("packedPFCandidates"),
+                                    cut=cms.string("fromPV(0) > 0")
+                                    )
+    task.add(process.CHSCands)
+
+    process.pfMetCHS = cms.EDProducer("PFMETProducer",
+                                      src = cms.InputTag("CHSCands"),
+                                      alias = cms.string('pfMet'),
+                                      globalThreshold = cms.double(0.0),
+                                      calculateSignificance = cms.bool(False),
+                                      )
+    task.add(process.pfMetCHS)    
+
+    addMETCollection(process,
+                     labelName = "patChsMet",
+                     metSource = "pfMetCHS"
+                     )
+
+    process.patChsMet.computeMETSignificance = cms.bool(False)
+
+    #  ==================  CHSMET 
+
+    #  ==================  TrkMET 
+    process.TrkCands = cms.EDFilter("CandPtrSelector",
+                                    src=cms.InputTag("packedPFCandidates"),
+                                    cut=cms.string("fromPV(0) > 0 && charge()!=0")
```

Comment:
@zdemirag I meant to comment on this one (tkMET) not on the one above (chsMET), for this I do not see any reason to be consistent with CHS, we should rather have something that has the features expected by the users (i.e. stability against PU) hence we could use a tighter PV association such as
`charge()!=0 && pvAssociationQuality()>=5 && vertexRef().key()==0`

Reply:
... or perhaps even `>=4`

Reply:
ah thats fine then. I thought you were referring to CHS. Give me (tops) one-two days, I have preliminary studies that can back up the decision we make. but in general, I see no problem updating this portion of the code in this PR and will update.

Reply:
i'll move the definition to ==> "charge()!=0 && pvAssociationQuality()>=4 && vertexRef().key()==0" , it is because >=4 has slightly lower energy removal for barrel jets (~5%) at all pts in a non-pu sample.

Reply:
@fgolf @peruzzim here the original discussion on why 4 vs 5

Reply:
I guess mini was updated after nano was already implemented